### PR TITLE
Create category APIs and refactor user module dto folders

### LINKED ISSUE #15: Category CRUD APIs
```
GET /categories
POST /categories
GET /categories/:id
POST /categories/:id
DELETE /categories/:id
```

### PR DESCRIPTION
**Pull request by PCalape**

### CODE REVIEW

File: src/modules/category/adapter/controller/category.controller.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @Post('/:id')
+    @UseGuards(SupabaseGuard)
+    async updateCategory(@Param('id') id: string, @Body() category: CategoryInput): Promise<void> {
+        await this.categoryService.updateCategory(id, category);
+    }
+
+    @Get()
+    @UseGuards(SupabaseGuard)
+    async fetchCategories(): Promise<CategoryOutput[]> {
+        return await this.categoryService.getCategories();
```
Comment (arnellebalane) [inline src/modules/category/adapter/controller/category.controller.ts:29]: @PCalape Can you temporarily make the other APIs not callable by anyone, for the initial version of the app we will create an initial list of categories that users can use. In the second version that's when we will allow users to create their own categories.

For the initial version, let's keep all these APIs but make sure that users can only call the `GET /categories` and `GET /categories/:id`  APIs.

**Review by arnellebalane — CHANGES_REQUESTED** [review]

File: src/modules/category/adapter/controller/category.controller.ts (hunk 1 above)
Comment (PCalape) [reply src/modules/category/adapter/controller/category.controller.ts:29]: Hello @arnellebalane sure will apply this one. Thank you!

Comment (PCalape) [reply src/modules/category/adapter/controller/category.controller.ts:29]: Hello @arnellebalane applied the suggestion and also refactored the environment variables for easier usage. Thanks!

File: src/modules/category/adapter/controller/category.controller.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { ENABLE_CATEGORY_MUTATION } from 'src/modules/common/environment';
+
+@Controller('/categories')
+export class CategoryController {
+    constructor(private readonly categoryService: CategoryService) {}
+
+    @Post()
+    @UseGuards(SupabaseGuard)
+    async createCategory(@Body() category: CategoryInput): Promise<void> {
+        if (!ENABLE_CATEGORY_MUTATION) throw new UnprocessableEntityException('Category mutation is currently disabled');
```
Comment (arnellebalane) [inline src/modules/category/adapter/controller/category.controller.ts]: @PCalape Can we make the API return a 404 response when this is still disabled?

File: src/modules/common/environment/index.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import * as dotenv from 'dotenv';
+
+dotenv.config();
+
+const SUPABASE_URL = process.env.SUPABASE_URL || '';
+const SUPABASE_KEY = process.env.SUPABASE_KEY || '';
+const SUPABASE_JWT_SECRET = process.env.SUPABASE_JWT_SECRET || '';
+const ENABLE_CATEGORY_MUTATION = process.env.ENABLE_CATEGORY_MUTATION === 'true';
+
+export {
```
Comment (arnellebalane) [inline src/modules/common/environment/index.ts]: @PCalape You can export variables using `export const VARIABLE_NAME = '...';`, is there a reason for exporting them like this?

File: src/modules/category/adapter/controller/category.controller.ts (hunk 2 above)
Comment (PCalape) [reply src/modules/category/adapter/controller/category.controller.ts]: Sure @arnellebalane will do

File: src/modules/common/environment/index.ts (hunk 3 above)
Comment (PCalape) [reply src/modules/common/environment/index.ts]: I think your suggestion is better @arnellebalane. No particular reason in my current implementation thanks!

File: src/modules/category/adapter/controller/category.controller.ts (hunk 2 above)
Comment (PCalape) [reply src/modules/category/adapter/controller/category.controller.ts]: Applied the changes @arnellebalane for your review thanks!

**Review by arnellebalane — APPROVED** [review]
LGTM 💯

**Pull request merged**